packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -31,11 +32,14 @@ export class FileTreeModel extends TreeModelImpl implements LocationService {
     @inject(FileSystem) protected readonly fileSystem: FileSystem;
     @inject(FileSystemWatcher) protected readonly watcher: FileSystemWatcher;
 
+    protected readonly _drives = new Deferred<URI[]>();
+
     @postConstruct()
     protected init(): void {
         super.init();
         this.toDispose.push(this.watcher.onFilesChanged(changes => this.onFilesChanged(changes)));
         this.toDispose.push(this.watcher.onDidMove(move => this.onDidMove(move)));
+        this.fileSystem.getDrives().then(drives => this._drives.resolve(drives.map(uri => new URI(uri))));
```

Comment:
'looking at the node filesystem implementation, `getDrives` may reject with error. it would be good to catch here.

Reply:
... and maybe to resolve the deferred with `[]`, no?